server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -754,6 +758,7 @@ public void onResponse(IngestPipelinesExecutionResult result) {
                                         if (result.shouldKeep == false) {
                                             onDropped.accept(slot);
                                         }
+                                        firstPipeline.getPipelineMetrics().postIngest(indexRequest.ramBytesUsed());
```

Comment:
I don't think it should be possible to reach that part of the code if the pipeline is null. The motivation behind adding the null check [earlier](https://github.com/elastic/elasticsearch/blob/43baaf9f231a07f97147998c25be858427a91093/server/src/main/java/org/elasticsearch/ingest/IngestService.java#L742) was because it's possible to receive a request containing a pipeline name that doesn't actually exist (there's a unit test that checks for this), and at that point `getFirstPipeline()` would return null. But at this stage we should have already successfully executed that pipeline so it can't have been null.

I can still add in the check here if you think that would be best?

Reply:
You are definitely right, I see the null check [here](https://github.com/elastic/elasticsearch/pull/107240/files#diff-055eff0044e6cbbab2fdd7c5a0fb5bf06afb55141c00c6d89c598ef6870e908aR906) guards this code. In that case, I don't think a check if needed, but maybe just an `assert firstPipeline != null` to show we thought of it!